tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,447 @@
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                            msg="Global signatureRequired policy is false, so signature should not be enforced.")
+
+    def test_should_enforce_signature_if_global_signatureRequired_true_and_individual_signatureRequired_not_specified(self):
+        """
+        If individual policy is present, but signatureRequired is not specified for that policy, use global signatureRequired.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": True,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertTrue(should_enforce_signature,
+                            msg="Individual signatureRequired policy is not set, so should use global policy and enforce signature.")
+
+    def test_should_not_enforce_signature_if_no_custom_policy_present(self):
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                         msg="No custom policy is present, so use default policy. Should not enforce signature.")
+
+    def test_should_allow_if_no_custom_policy_present(self):
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine()
+        should_allow = engine.should_allow_extension(test_extension)
+        self.assertTrue(should_allow,
+                        msg="No custom policy is present, so use default policy. Should allow all extensions.")
+
+    def test_should_allow_if_extension_policy_section_missing(self):
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0"
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_allow = engine.should_allow_extension(test_extension)
+        self.assertTrue(should_allow)
+
+    def test_should_allow_if_policy_disabled(self):
```

Comment:
__get_policy_enforcement_enabled() checks if the policy file exists first. Since policy file doesn't exist, the function is returning false without checking the conf flag. Consider creating a policy file for this test and test_should_not_enforce_signature_if_policy_disabled

Reply:
Refactored these into two different tests:
test_should_allow_and_should_not_enforce_signature_if_no_custom_policy_file
test_should_allow_and_should_not_enforce_signature_if_conf_flag_false